prisma-fmt/src/code_actions/relations.rs

```diff
@@ -180,13 +194,109 @@ pub(super) fn add_referenced_side_unique(
     actions.push(CodeActionOrCommand::CodeAction(action));
 }
 
-fn create_missing_unique<'a>(
+/// For schema's with emulated relations,
+/// If the referenced side of the relation does not point to a unique
+/// constraint, the action adds the attribute.
+///
+/// If referencing a single field:
+///
+/// ```ignore
+/// model A {
+///     id      Int @id
+///     field1  B   @relation(fields: [bId], references: [id])
+///                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ // Warn
+///     bId     Int
+///
+///     // <- suggest @@index([bId]) here
+/// }
+///
+/// model B {
+///     id Int @id
+///     as A[]
+/// }
+/// ```
+///
+/// If referencing multiple fields:
+///
+/// ```ignore
+/// model A {
+///     id      Int @id
+///     field1  B   @relation(fields: [bId1, bId2, bId3], references: [id1, id2, id3])
+///                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ // Warn
+///     bId1    Int
+///     bId2    Int
+///     bId3    Int
+///
+///     // <- suggest @@index([bId1, bId2, bId3]) here
+/// }
+///
+/// model B {
+///     id1 Int
+///     id2 Int
+///     id3 Int
+///     as  A[]
+///
+///     @@id([id1, id2, id3])
+/// }
+/// ```
+pub(super) fn add_index_for_relation_fields(
+    actions: &mut Vec<CodeActionOrCommand>,
+    params: &CodeActionParams,
+    schema: &str,
+    relation: RelationFieldWalker<'_>,
+) {
+    let Some(fields) = relation.fields() else { return; };
+    if relation.model().indexes().any(|index| {
+        index
+            .fields()
+            .zip(fields.clone())
+            .all(|(index_field, relation_field)| index_field.field_id() == relation_field.field_id())
+    }) {
+        return;
+    }
+
+    let attribute_name = "index";
+    let (new_text, range) = create_block_attribute(schema, relation.model(), fields, attribute_name);
+    let text = TextEdit { range, new_text };
+
+    let mut changes = HashMap::new();
+    changes.insert(params.text_document.uri.clone(), vec![text]);
+
+    let edit = WorkspaceEdit {
+        changes: Some(changes),
+        ..Default::default()
+    };
+
+    let Some(span_diagnostics) = super::diagnostics_for_span(
+        schema,
+        &params.context.diagnostics,
+        relation.relation_attribute().unwrap().span(),
+    ) else { return; };
+
+    let diagnostics = span_diagnostics
+        .into_iter()
+        .filter(|diag| diag.message.contains("relationMode = \"prisma\""))
```

Comment:
Unsure if something we already have available -- if not, it would be a nice future tech upgrade for our diagnostics -- but there are other fields that would be nice here. Specifically looking at Diagnostic's `code`